### PR TITLE
Initial version of sbt-javaagent

### PR DESCRIPTION
**Pull request by pvlugter**

### CODE REVIEW

File: src/main/scala/com/lightbend/sbt/javaagent/JavaAgentPackaging.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      val name = if (className.endsWith("$")) className else (className + "$")
+      Class.forName(name).asInstanceOf[Class[AutoPlugin]]
+        .getDeclaredField("MODULE$").get(null) match {
+          case plugin: AutoPlugin => plugin
+          case _ => throw new RuntimeException(s"[$name] is not an AutoPlugin object")
+        }
+    } catch {
+      case _: Throwable => NoPlugin
+    }
+  }
```
Comment (bantonsson) [inline src/main/scala/com/lightbend/sbt/javaagent/JavaAgentPackaging.scala]: I love this workaround. Very Nice!

File: maxwell/src/main/java/maxwell/Maxwell.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package maxwell;
+
+import java.lang.instrument.Instrumentation;
+
+public class Maxwell {
+    public static void premain(String agentArgs, Instrumentation instrumentation) {
+        System.out.println("Agent 86");
```
Comment (bantonsson) [inline maxwell/src/main/java/maxwell/Maxwell.java]: Trying to be Smart are we? 😉

File: src/main/scala/com/lightbend/sbt/javaagent/JavaAgent.scala [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  override def projectSettings = Seq(
+    javaAgents := Seq.empty,
+    ivyConfigurations += AgentConfig,
+    libraryDependencies ++= javaAgents.value.map(_.module),
+    resolvedJavaAgents := resolveAgents.value,
+    fork in run := enableFork(fork in run, _.scope.run).value,
+    fork in Test := enableFork(fork in Test, _.scope.test).value,
+    javaOptions in run ++= agentOptions(_.agent.scope.run).value,
+    javaOptions in Test ++= agentOptions(_.agent.scope.test).value
```
Comment (bantonsson) [inline src/main/scala/com/lightbend/sbt/javaagent/JavaAgent.scala]: So won't the agent jar end up twice on the class path if you add it to `"compile"`or `"provided"` and `"test"` scopes? I mean once in the `libraryDependencies` and once in the `agentOptions`?

**Comment by bantonsson** [comment]
LGTM

File: src/main/scala/com/lightbend/sbt/javaagent/JavaAgent.scala (hunk 3 above)
Comment (pvlugter) [inline src/main/scala/com/lightbend/sbt/javaagent/JavaAgent.scala]: Yes, for `test` it ends up being on both the classpath and as an agent. `run` and `dist` have classpaths that don't include `compile` or `provided`. Not sure if there's a nice way around this, apart from changing the fork test directly, to filter the classpath. I was thinking that we just have the double up for tests. It shouldn't cause any issues? While `run` and `dist` use a proper provided at runtime approach.

File: maxwell/src/main/java/maxwell/Maxwell.java (hunk 2 above)
Comment (pvlugter) [inline maxwell/src/main/java/maxwell/Maxwell.java]: Maybe the Cinnamon agent should be called Maxwell?

**Comment by pvlugter** [comment]
I'll merge and publish this first version now. We can figure out test classpath separately.

**Pull request merged**